config/loader_test.go

```diff
@@ -0,0 +1,151 @@
+package config_test
+
+import (
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/odpf/optimus/config"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	configFileName       = ".optimus.yaml"
+	optimusConfigDirName = "./optimus"
+	optimusConfigContent = `
+version: 1
+log:
+  level: info
+host: "localhost:9100"
+project:
+  name: sample_project
+  config:
+    environment: integration
+    scheduler_host: http://example.io/
+    storage_path: file://absolute_path_to_a_directory
+serve:
+  port: 9100
+  host: localhost
+  ingress_host: optimus.example.io:80
+  app_key: Yjo4a0jn1NvYdq79SADC/KaVv9Wu0Ffc
+  replay_num_workers: 1
+  replay_worker_timeout: "100s"
+  replay_run_timeout: "10s"
+  db:
+    dsn: postgres://user:password@localhost:5432/database?sslmode=disable
+    max_idle_connection: 5
+    max_open_connection: 10
+scheduler:
+  name: airflow2
+  skip_init: true
+telemetry:
+  profile_addr: ":9110"
+  jaeger_addr: "http://localhost:14268/api/traces"
+namespaces:
+- name: namespace-a
+  job:
+    path: ./jobs-a
+- name: namespace-b
+  job:
+    path: ./jobs-b
+`
+)
+
+func setup(content string) {
+	teardown()
+	if err := os.Mkdir(optimusConfigDirName, os.ModePerm); err != nil {
```

Comment:
`ModePerm` is `0o777`, not good for file permissions. Should use more restrictive permissions.

address in next pr.

Reply:
updated to put more restriction on the permission